### PR TITLE
Repair template_vars functionality and allow for dest keyword in rpc on Junos

### LINKED ISSUE #57388: [BUG] Junos execution modules ignoring template_vars on load and diff extra keyword
**Description**
Recent changes removed template_vars breaking load functionality.
In addition supplying a dest parameter to rpc call generates errors, that is, junos rejects extraneous keyword

**Setup**
State file contents
```
salt://files/rest.set:
    junos:
       - load
       - timeout: 100 
       - template_vars:
           mgmtip: 192.168.56.100
```
where salt://files/rest.set contains:
```
set system services rest control allowed-sources [ 127.0.0.1 {{ template_vars["mgmtip"] }} ]
```

**Steps to Reproduce the behavior**
```
root@Unknown:/srv/salt# salt junos174RS8 junos.load "salt://files/rest.set" template_vars='{"mgmtip": "192.168.56.51"}'
junos174RS8:
    ----------
    message:
        Salt failed to render the template, please check file path and syntax.
        Error: Jinja variable 'template_vars' is undefined
    out:
        False
```
```
root@Unknown:/srv/salt# salt junos174RS8 junos.rpc "get-software-information" dest=/tmp/rpc.out
junos174RS8:
    ----------
    message:
        RPC execution failed due to "RpcError(severity: error, bad_element: dest, message: error: syntax error, expecting <invoke-on>
        error: syntax error)"
    out:
        False
```
**Expected behavior**
```
root@Unknown:/srv/salt# salt junos174RS8 junos.load "salt://files/rest.set" template_vars='{"mgmtip": "192.168.56.51"}'
junos174RS8:
    ----------
    message:
        Successfully loaded the configuration.
    out:
        True
```

```
root@Unknown:/srv/salt# salt junos174RS8 junos.rpc "get-software-information" dest=/tmp/rpc.out
junos174RS8:
    ----------
    out:
        True
    rpc_reply:
        ----------
        software-information:
            ----------
            host-name:
                junos174RS8
            junos-version:
                17.4R2-S8
            package-information: ....
```
**Screenshots**

**Versions Report**
Current native minion utilizing master branch code, merging fix back into master branch

**Additional context**

### PR DESCRIPTION
**Pull request by dmurphy18**
### What does this PR do?
Repairs template_vars functionality on Junos and compensates for dest keyword in rpc commands

### What issues does this PR fix or reference?
Fixes: https://github.com/saltstack/salt/issues/57388

### Previous Behavior
load and certain rpc's on Junos would fail, see https://github.com/saltstack/salt/issues/57388

### New Behavior
load and rpc commands with dest parameters now function correctly, see https://github.com/saltstack/salt/issues/57388

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [X] Docs
- [X] Changelog - https://docs.saltstack.com/en/master/topics/development/changelog.html
- [X] Tests written/updated

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Comment by dmurphy18** [comment]
@vnitinv Can you take a look at this too and leave comments

File: salt/modules/junos.py
Comment (vnitinv) [inline salt/modules/junos.py:226]: this need change. op will contain dest if called via state file.
so it should be 

if dest is None:
    dest = op.pop("dest", None)

or we can also do
dest = op.pop("dest", dest)

Comment (dmurphy18) [reply salt/modules/junos.py:226]: Found that if dest is specified via a state file or execution module that dest is there as a keyword.

**Review by s0undt3ch — DISMISSED** [review]

**Review by twangboy — DISMISSED** [review]

**Review by dwoz — DISMISSED** [review]

**Review by dwoz — DISMISSED** [review]

**Comment by dmurphy18** [comment]
Re-did changes after an updated merge of master branch was done out of order for these changes

**Comment by vnitinv** [comment]
I am discussing this with @dmurphy18 , please do not merge it.

**Comment by dmurphy18** [comment]
Discussed issue with Nitin and new changes made to resolve concerns, ready for review and merge

**Review by twangboy — APPROVED** [review]

**Review by dwoz — APPROVED** [review]

**Pull request merged**
